### PR TITLE
Small fixes and additions to the security best practices

### PR DESCRIPTION
**Pull request by gwillem**
## Purpose of this pull request

This pull request fixes a few typos and adds common security best practices.

## Magento release version

Applies to all versions.

## Product editions

Is this update specific to a product edition?

- [x] Magento Open Source only
- [x] Magento Commerce only

Is this update specific to an installed feature extension

No

## Affected documentation pages

- https://docs.magento.com/user-guide/magento/magento-security-best-practices.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: src/magento/magento-security-best-practices.md [hunk 1]
```diff
@@ -23,29 +27,29 @@ Protecting the environment is the most critical aspect of ensuring the security
 
 ### Server environment
 
-- Make sure that the server operating system is secure. Work with your hosting provider to ensure that  there is no unnecessary software running on the server.
+- Make sure that the server operating system is secure. Work with your hosting provider to ensure that there is no unnecessary software running on the server.
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:30]:
```suggestion
- Make sure that the server operating system is secure. Work with your hosting provider to ensure that there is no unnecessary software running on the server.
```
```suggestion
- Make sure that the server operating system is secure. Work with your hosting provider to ensure that there is no unnecessary software running on the server.
```
```suggestion
- Make sure that the server operating system is secure. Work with your hosting provider to ensure that no unnecessary software is running on the server.
```

File: src/magento/magento-security-best-practices.md [hunk 2]
```diff
@@ -23,29 +27,29 @@ Protecting the environment is the most critical aspect of ensuring the security
...
 - Use only secure communications protocol (SSH/SFTP/HTTPS) to manage files, and disable FTP.
-- Magento includes `.htaccess` files to protect system files when using the Apache web server. If you use a different web server such as NGINX, make sure that all system files and directories are protected. For an sample NGINX configuration, see [magento-nginx.conf](https://gist.github.com/gwillem/cd5ae6845fa33aa0d481){:target="_blank"} on GitHub.
-- Use strong and unique passwords, and change them periodically.
+- Magento includes `.htaccess` files to protect system files when using the Apache web server. If you use a different web server such as NGINX, make sure that all system files and directories are protected. For a sample NGINX configuration, see [magento-nginx.conf](https://gist.github.com/gwillem/cd5ae6845fa33aa0d481){:target="\_blank"} on GitHub.
+- Use strong and unique passwords, and change them periodically. Strong means: computer generated and a minimum of 8 characters.
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:33]:
```suggestion
- Use strong and unique passwords with a minimum of eight characters, and change them periodically.  
```

File: src/magento/magento-security-best-practices.md [hunk 3]
```diff
@@ -23,29 +27,29 @@ Protecting the environment is the most critical aspect of ensuring the security
...
-- Use two-factor authorization for Admin logins. There are several extensions available that provide additional security by requiring an additional passcode that is generated on your phone, or a token from a special device.
-- Review your server for “development leftovers.” Make sure there are no accessible log files, publicly visible .git directories, tunnels to execute SQL, database dumps, `phpinfo` files, or any other unprotected files that are not required, and that might be used in an attack.
+- Limit access to the Magento Admin by updating the allow list with the IP address of each computer that is authorized to use the Admin and Magento Connect downloader. For examples of how to add IP addresses to the allow list, see [Secure Your Magento Admin](https://www.sonassi.com/knowledge-base/magento-kb/secure-your-magento-admin/){:target="\_blank"}.
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:41]:
```suggestion
- Limit access to the Magento Admin by updating the allow list with the IP address of each computer that is authorized to use the Admin. For examples of how to add IP addresses to the allow list, see [Secure Your Magento Admin](https://www.sonassi.com/knowledge-base/magento-kb/secure-your-magento-admin/){:target="\_blank"}.
```

File: src/magento/magento-security-best-practices.md [hunk 4]
```diff
@@ -23,29 +27,29 @@ Protecting the environment is the most critical aspect of ensuring the security
...
+- Do not install extensions directly on a production server. To disable the Magento Connect downloader on the production site, either remove or block access to the `/downloader` directory. You can also use the same methods to [allow access](https://www.sonassi.com/knowledge-base/magento-kb/secure-your-magento-admin/){:target="\_blank"}.
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:42]:
```suggestion
- Do not install extensions directly on a production server. 
```

Comment (ccondra) [reply src/magento/magento-security-best-practices.md:42]: The Magento Connect downloader is no longer available.

File: src/magento/magento-security-best-practices.md (hunk 3 above)
Comment (ccondra) [reply src/magento/magento-security-best-practices.md:41]: The Magento Connect downloader is no longer used.

File: src/magento/magento-security-best-practices.md [hunk 5]
```diff
@@ -23,29 +27,29 @@ Protecting the environment is the most critical aspect of ensuring the security
...
 - Limit outgoing connections to only those that are required, such as for a payment integration.
 - Use a Web Application Firewall to analyze traffic and discover suspicious patterns, such as credit card information being sent to an attacker.
 
 ### Server applications
 
-- Make sure that all applications running on the server are secure.
+- Make sure that all applications running on the server are secure. Keep all software up to date, and apply patches as recommended.
 - Avoid running other software on the same server as Magento, especially if it is accessible from the Internet. Vulnerabilities in blog applications such as Wordpress can expose private information from Magento. Install such software on a separate server or virtual machine.
-- Keep all software  up to date, and apply patches as recommended.
+- Do not install web-based database managers (such as PHPmyAdmin or Adminer) on your production server, as they can be used as backdoor by attackers.
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:52]:
```suggestion
- Do not install third-party web-based database managers on your production server, as they can provide backdoor access to attackers.
```

File: src/magento/magento-security-best-practices.md [hunk 6]
```diff
@@ -62,11 +66,11 @@ Your effort to protect your Magento installation starts with the initial setup,
...
-    {:.bs-callout-info}
-    Check with your hosting provider before implementing a custom Admin URL. Some hosting providers require a standard URL to meet firewall protection rules.
+  {:.bs-callout-info}
+  Check with your hosting provider before implementing a custom Admin URL. Some hosting providers require a standard URL to meet firewall protection rules.
 - Block access to any development, staging, or testing systems. Use IP [allow lists](https://www.sonassi.com/knowledge-base/magento-kb/secure-your-magento-admin/) and `.htaccess` password protection. When compromised, such systems can produce a data leak or be used to attack the production system.
 - Use the correct file permissions. Core Magento and directory files should be set to ready only, including `app/etc/local.xml` files.
-- Use a strong  password for the Magento Admin. To learn more, see [Creating a strong password](https://support.google.com/accounts/answer/32040?hl=en).
+- Use a strong password for the Magento Admin. To learn more, see [Creating a strong password](https://support.google.com/accounts/answer/32040?hl=en).
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:73]:
```suggestion
- Use a [strong](https://support.google.com/accounts/answer/32040?hl=en) password for the Magento Admin.
```

File: src/magento/magento-security-best-practices.md [hunk 7]
```diff
@@ -88,11 +92,12 @@ If your system is not immediately patched after a major security breach, there i
 
 ### Security review
 
+- Run a server-side [Magento malware and vulnerability monitor][12] to quickly alert you of security issues with your files, database, admin accounts and third party components.
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:95]:
```suggestion
- Run a server-side [Magento malware and vulnerability monitor][12] to quickly alert you of security issues with your files, database, admin accounts, and third party components.
```

File: src/magento/magento-security-best-practices.md [hunk 8]
```diff
@@ -23,29 +27,29 @@ Protecting the environment is the most critical aspect of ensuring the security
...
+- Limit access to the Magento Admin by updating the allow list with the IP address of each computer that is authorized to use the Admin and Magento Connect downloader. For examples of how to add IP addresses to the allow list, see [Secure Your Magento Admin](https://www.sonassi.com/knowledge-base/magento-kb/secure-your-magento-admin/){:target="\_blank"}.
+- Do not install extensions directly on a production server. To disable the Magento Connect downloader on the production site, either remove or block access to the `/downloader` directory. You can also use the same methods to [allow access](https://www.sonassi.com/knowledge-base/magento-kb/secure-your-magento-admin/){:target="\_blank"}.
+- Use two-factor authorization for Admin logins, which is included by default in Magento 2.4. For older releases, there are several extensions available that provide additional security by requiring an additional passcode that is generated on your phone, or a token from a special device.
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:43]:
```suggestion
- Use two-factor authorization for Admin logins, as required by Magento 2.4.x. For older releases, there are several extensions available that provide additional security and require a one-time passcode that is generated on your phone, or a token from a special device.
```

File: src/magento/magento-security-best-practices.md [hunk 9]
```diff
@@ -23,29 +27,29 @@ Protecting the environment is the most critical aspect of ensuring the security
...
+- Review your server for “development leftovers.” Make sure there are no accessible log files, publicly visible .git directories, tunnels to execute SQL, database dumps, backup archives, `phpinfo` files, or any other unprotected files that are not required, and that might be used in an attack.
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:44]:
```suggestion
- Review your server for “development leftovers.” Make sure there are no accessible log files, publicly visible .git directories, tunnels to execute SQL, database dumps, backup archives, `phpinfo` files, or any other unprotected files that are not required and might be used in an attack.
```

**Review by ccondra — COMMENTED** [review]
Hi -- I made a few comments and suggestions that will be reviewed by our security team lead. Thank you!  -- Claire

**Review by ccondra — APPROVED** [review]
Thank you!

File: src/magento/magento-security-best-practices.md [hunk 10]
```diff
@@ -23,29 +27,29 @@ Protecting the environment is the most critical aspect of ensuring the security
...
+- Make sure that the server operating system is secure. Work with your hosting provider to ensure that no unnecessary software is running on the server.
 - Use only secure communications protocol (SSH/SFTP/HTTPS) to manage files, and disable FTP.
-- Magento includes `.htaccess` files to protect system files when using the Apache web server. If you use a different web server such as NGINX, make sure that all system files and directories are protected. For an sample NGINX configuration, see [magento-nginx.conf](https://gist.github.com/gwillem/cd5ae6845fa33aa0d481){:target="_blank"} on GitHub.
-- Use strong and unique passwords, and change them periodically.
+- Magento includes `.htaccess` files to protect system files when using the Apache web server. If you use a different web server such as NGINX, make sure that all system files and directories are protected. For a sample NGINX configuration, see [magento-nginx.conf](https://gist.github.com/gwillem/cd5ae6845fa33aa0d481){:target="\_blank"} on GitHub.
+- Use [strong](https://support.google.com/accounts/answer/32040?hl=en), unique passwords with a minimum of eight characters, and change them periodically.
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:33]:
```suggestion
- Use [strong](https://support.google.com/accounts/answer/32040?hl=en), unique passwords with a minimum of eight characters, and change them periodically.
```

Comment (ccondra) [reply src/magento/magento-security-best-practices.md:33]: Removed trailing spaces

File: src/magento/magento-security-best-practices.md [hunk 11]
```diff
@@ -23,29 +27,29 @@ Protecting the environment is the most critical aspect of ensuring the security
...
-- Use two-factor authorization for Admin logins. There are several extensions available that provide additional security by requiring an additional passcode that is generated on your phone, or a token from a special device.
-- Review your server for “development leftovers.” Make sure there are no accessible log files, publicly visible .git directories, tunnels to execute SQL, database dumps, `phpinfo` files, or any other unprotected files that are not required, and that might be used in an attack.
+- Limit access to the Magento Admin by updating the allow list with the IP address of each computer that is authorized to use the Admin and install extensions. For examples of how to add IP addresses to the allow list, see [Secure Your Magento Admin](https://www.sonassi.com/knowledge-base/magento-kb/secure-your-magento-admin/){:target="\_blank"}.
+- Do not install extensions directly on a production server.
```
Comment (ccondra) [inline src/magento/magento-security-best-practices.md:42]:
```suggestion
- Do not install extensions directly on a production server.
```

Comment (ccondra) [reply src/magento/magento-security-best-practices.md:42]: Removed trailing space.

**Review by ccondra — APPROVED** [review]
Removed a few trailing spaces.

**Comment by ccondra** [comment]
running tests

**Comment by gwillem** [comment]
Thanks for maintaining these docs!

**Pull request merged**